lib/pipeline.js

```diff
@@ -497,8 +497,8 @@ class PipelineModel extends BaseModel {
 
         return factory.list(listConfig)
             .then((jobs) => {
-                // If archived is true, don't sort and just return jobs
-                if (listConfig.params.archived) {
+                // If using the new config or archived is true, don't sort and just return jobs
+                if (!workflow || workflow.length === 0 || listConfig.params.archived) {
```

Comment:
it could be `!Array.isArray(worklflow) || workflow.length`

Reply:
Nice!